pkg/kubectl/resource/builder.go

```diff
@@ -806,12 +806,13 @@ func HasNames(args []string) (bool, error) {
 
 // MultipleTypesRequested returns true if the provided args contain multiple resource kinds
 func MultipleTypesRequested(args []string) bool {
+	if len(args) == 1 && args[0] == "all" {
+		return true
```

Comment:
> this check is mostly for cases such as https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/cmd/get.go#L424

Did you mean this line? (line number was changed since the link points to `master`):

```go
if resource.MultipleTypesRequested(args) || cmdutil.MustPrintWithKinds(objs, infos, sorter) {
```

IIUC, before this line, we make a call to `ResourceTypeOrNameArgs`:

```go 
	r := resource.NewBuilder(mapper, typer, resource.ClientMapperFunc(f.UnstructuredClientForMapping), unstructured.UnstructuredJSONScheme).
		NamespaceParam(cmdNamespace).DefaultNamespace().AllNamespaces(allNamespaces).
		FilenameParam(enforceNamespace, &options.FilenameOptions).
		SelectorParam(selector).
		ExportParam(export).
		ResourceTypeOrNameArgs(true, args...).
		ContinueOnError().
		Latest().
		Flatten().
		Do()
```

After this call, `args` has been changed (aliases, including `all`, are expanded). Would you explain more on when the following situation would happen? 
> arguments from a sub command are passed to it, without first going through other functions in resource/builder.go